lib/committee/query_hash_coercer.rb

```diff
@@ -0,0 +1,35 @@
+module Committee
+  class QueryHashCoercer
+    def initialize(query_hash, schema)
+      @query_hash = query_hash
+      @schema = schema
+    end
+
+    def call
+      coerced = {}
+      @schema.properties.each do |k, s|
+        original_val = @query_hash[k]
+        unless original_val.nil?
+          s.type.each do |to_type|
+            begin
+              case to_type
+              when "null"
+                coerced[k] = nil if original_val.empty?
+              when "integer"
+                coerced[k] = Integer(original_val)
+              when "number"
+                coerced[k] = Float(original_val)
+              when "boolean"
+                coerced[k] = true if original_val == "true"
+                coerced[k] = false if original_val == "false"
+              end
+            rescue ArgumentError
```

Comment:
It sucks that these errors are so unspecific, but can we conditionally rescue here only if the message matches "invalid value for"? I've been bitten too many times by broad rescue statements that end up doing the wrong thing.

It might actually be a good idea to move the rescue specifically into the "integer" and "number" blocks and then match against something like "invalid value for Integer()", just for maximum certainty. The test suite should give us protection against the error message changing in the future.